AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/DeviceRun.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.androidx.ci.testRunner
+
+/**
+ * Represents a device that run a test
+ */
+data class DeviceRun internal constructor(
+    /**
+     * The device identifier that run the test.
+     * e.g. redfin-30-en-portrait
+     * e.g. redfin-30-en-portrait_rerun_1
+     * e.g. redfin-30-en-portrait-shard_0
+     * e.g. redfin-30-en-portrait-shard_2-rerun_2
+     */
+    val fullDeviceId: String,
```

Comment:
this sounds a bit more like uniqueRunId or something like that.

Reply:
or maybe just uniqueId

Reply:
yea i've struggled w/ this.(and class name). The problem with uniqueId is that it is only unique within a test matrix so that might be confusing. 
I can call it `id` ?